crates/factors-executor/src/lib.rs

```diff
@@ -0,0 +1,167 @@
+use std::collections::HashMap;
+
+use anyhow::Context;
+use spin_app::{App, AppComponent};
+use spin_core::Component;
+use spin_factors::{AsInstanceState, ConfiguredApp, RuntimeFactors, RuntimeFactorsInstanceState};
+
+/// A FactorsExecutor manages execution of a Spin app.
+///
+/// `Factors` is the executor's [`RuntimeFactors`]. `ExecutorInstanceState`
+/// holds any other per-instance state needed by the caller.
+pub struct FactorsExecutor<Factors: RuntimeFactors, ExecutorInstanceState = ()> {
+    factors: Factors,
+    core_engine: spin_core::Engine<InstanceState<Factors::InstanceState, ExecutorInstanceState>>,
+    configured_app: ConfiguredApp<Factors>,
+    component_instance_pres: HashMap<String, InstancePre<Factors, ExecutorInstanceState>>,
```

Comment:
Nit: can you document what `component_instance_pres` is? At the very least it would be nice to know what the keys are without having to read the code.

Reply:
done